python/ray/autoscaler/azure/node_provider.py

```diff
@@ -9,10 +9,10 @@
 from azure.mgmt.compute.models import ResourceIdentityType
 
 from ray.autoscaler.node_provider import NodeProvider
-from ray.autoscaler.tags import TAG_RAY_CLUSTER_NAME, TAG_RAY_NODE_NAME
+from ray.autoscaler.tags import TAG_RAY_CLUSTER_NAME, TAG_RAY_NODE_NAME, TAG_RAY_NODE_TYPE, NODE_TYPE_HEAD
 
 VM_NAME_MAX_LEN = 64
-VM_NAME_UUID_LEN = 4
+VM_NAME_UUID_LEN = 8
```

Comment:
do we really need to support more then 65,536 nodes?

Reply:
I'm more worried about collisions for the guids.

Reply:
fair